### ISSUE TITLE
Can't resume a SoundCloud playback

### ISSUE THREAD

**Issue by gardar** [tizonia/tizonia-openmax-il#437]
The soundcloud stream does not seem to continue if I pause it for a while (about an hour or so).

When I pause a soundcloud stream and then resume it the playback continues for a little while (probably when playing the cache) but then the audio stops, the playback does not continue to the next track, it does not stop or exit tizonia and I see no indication of what happened, it just seems to hang.

If I pause for a short while (within 5 minutes), and then resume then I don't encounter this issue.

I'm not sure if this also happens with other services such as youtube or if this is just related to soundcloud.

**Comment by tizonia** [comment]
Hi!, thanks for trying the software and for the feedback.

Some of the cloud services provide stream URLs that are valid for a limited amount of time. I don't remember exactly, but SoundCloud could be one of these cases.

Would you be able to share the soundcloud command that would reproduce the issue? With that, I should be able to make modifications to make Tizonia do something sensible (either exiting with an error message or perhaps internally re-issuing the query to retrieve a fresh set of URLs).

**Comment by gardar** [comment]
Hi, you can for example try

`tizonia --soundcloud-creator rinsefm`

the urls expiring does make sense and if we can get a fresh set of urls then that would be great, better than the alternative of increasing the buffer.

**Comment by kingo55** [comment]
Perhaps related to this issue, I will be playing a track from Soundcloud (often 1 or 2 hours long) and:

 - Playback is fine for quite a bit of the track
 - At some point the music stops (but the track progress continues playing)
 - Pausing and resuming does nothing, likewise muting unmuting...
 - Playback only works if I move to the next track, or go to the previous track

However then I cannot seek back to the original position.

**Comment by tizonia** [comment]
@kingo55 

Yes, it sounds like this is the same issue. URLs becoming invalid after some time. And seek is not currently implemented in Tizonia. I know these are two big issues for people trying to use Tizonia for listening to podcasts and shows. 

I don't currently have a timeline for these two as it is quite a bit of work. But any contributions on this and #438 would definitely be most welcome.

**Comment by fluential** [comment]
Still happening for me on 0.22, playing soundcloud longs tracks does not progress to the next.
